pkg/dashboard/resource/function.go

```diff
@@ -501,32 +501,29 @@ func (fr *functionResource) patchFunctionDesiredState(request *http.Request,
 func (fr *functionResource) redeployFunction(request *http.Request,
```

Comment:
you mixed the what and how on this function
you assume here you are using k8s platform. the how - you should not care of, this is up to the platform. 
I think better of use platform.RedeployFunction and just let the platform decide how it treat it internally
kube - change crd state
local - do the deploy

Reply:
I agree with letting the platform decide on the how, but for the local platform doing the actual deployment I would prefer to reuse the `functionResource`'s `storeAndDeployFunction` - which does extra enrichment before creating the function.
That would require the resource to be aware of the platform kind, which is also against the interface idea...